### PR TITLE
storyboook addon knobs select: correct types for an array

### PR DESCRIPTION
**Pull request by hipstersmoothie**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/storybook__addon-knobs/index.d.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ export function object<T>(name: string, value: T, groupId?: string): T;
 export function radios<T>(name: string, options: { [s: string]: T }, value?: T, groupId?: string): T;
 
 export function select<T>(name: string, options: { [s: string]: T }, value: T, groupId?: string): T;
+export function select<T>(name: string, options: ReadonlyArray<T>, value: T, groupId?: string): T;
```
Comment (joscha) [inline types/storybook__addon-knobs/index.d.ts]: @hipstersmoothie can you link me to the docs describing this interface, please?

**Comment by hipstersmoothie** [comment]
https://github.com/storybookjs/storybook/tree/master/addons/knobs#select

File: types/storybook__addon-knobs/index.d.ts (hunk 1 above)
Comment (hipstersmoothie) [reply types/storybook__addon-knobs/index.d.ts]: https://github.com/storybookjs/storybook/tree/master/addons/knobs#select

Comment (hipstersmoothie) [reply types/storybook__addon-knobs/index.d.ts]: > You can also provide options as an array like this: ['red', 'blue', 'yellow']

Comment (joscha) [reply types/storybook__addon-knobs/index.d.ts]: Yep, saw it in https://github.com/storybookjs/storybook/blob/master/addons/knobs/README.md#select - not sure why I missed that before. Do you mind adding an example in the tests?

Comment (hipstersmoothie) [reply types/storybook__addon-knobs/index.d.ts]: Sure!

Comment (hipstersmoothie) [reply types/storybook__addon-knobs/index.d.ts]: @joscha I have added a test that reflects how I use a select the most. Without my change the test fails ✅

**Review by joscha — APPROVED** [review]

**Comment by orta** [comment]
Cool, looks good - FWIW, does this only exist for older projects? the module is now built [in typescript](https://github.com/storybookjs/storybook/tree/master/addons/knobs/src)

**Pull request merged**